rag_experiment_accelerator/config/tests/test_config.py

```diff
@@ -32,10 +32,20 @@ def test_config_init(mock_validate_json_with_schema, mock_create_embedding_model
     embedding_model_1 = MagicMock()
     embedding_model_1.model_name.return_value = "all-MiniLM-L6-v2"
     embedding_model_1.dimension.return_value = 384
+    embedding_model_1.shorten_dimensions.return_value = False
     embedding_model_2 = MagicMock()
     embedding_model_2.model_name.return_value = "text-embedding-ada-002"
     embedding_model_2.dimension.return_value = 1536
-    mock_create_embedding_model.side_effect = [embedding_model_1, embedding_model_2]
+    embedding_model_2.shorten_dimensions.return_value = False
+    embedding_model_3 = MagicMock()
+    embedding_model_3.model_name.return_value = "text-embedding-3-large"
+    embedding_model_3.dimension.return_value = 3072
+    embedding_model_3.shorten_dimensions.return_value = False
+    embedding_model_4 = MagicMock()
+    embedding_model_4.model_name.return_value = "text-embedding-3-small"
+    embedding_model_4.dimension.return_value = 256
+    embedding_model_4.shorten_dimensions.return_value = True
+    mock_create_embedding_model.side_effect = [embedding_model_1, embedding_model_2, embedding_model_3, embedding_model_4]
     mock_validate_json_with_schema.return_value = (True, None)
```

Comment:
Severity: Medium - The `mock_create_embedding_model.side_effect` now includes multiple models. Consider creating a factory function to generate these models dynamically. This will reduce redundancy and improve maintainability as the number of models grows.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or know how to make it better and want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.